java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -354,4 +354,137 @@ public void negative() throws IOException {
       assert_().fail("Incorrect parameters passed to the checker");
     }
   }
+
+  @Test
+  public void positiveSpecificTreatmentGroupWithFlagNameAsVariable() throws IOException {
```

Comment:
Can these two test cases be simplified (we don't need to test for groups for these tests)? The check can simply be 
```
if(experimentation.isToggleEnabled(STALE_FLAG_CONSTANTS)) {
        return "X";
}
```
and 

```
if(experimentation.isToggleEnabled("STALE_FLAG")) {
        return "X";
}
```

You can look at more examples [here](https://github.com/uber/piranha/blob/master/java/piranha/src/test/resources/com/uber/piranha/XPFlagCleanerPositiveCases.java). 

Also, it may be easier to view the refactorings if they are added to that file and related files in this [dir](https://github.com/uber/piranha/tree/master/java/piranha/src/test/resources/com/uber/piranha).

Reply:
Also, couple of tests to check for the non-equivalent flag case. 

```
if(experimentation.isToggleEnabled("NOT_STALE_FLAG")) {
        return "Y";
}
```
should not be refactored.

Reply:
Right. These tests are testing the new change in conjunction with the more complex functionality about specific treatment groups. Not necessarily bad to have, but I'd rather have unit tests for the common case for variable and literal flags mentioned by Murali above :)

Reply:
fixed the suggestions